lib/semantics/check-do-stmt.cc

```diff
@@ -498,21 +595,15 @@ class DoConcurrentContext {
     CheckDefaultNoneImpliesExplicitLocality(localitySpecs);
   }
 
+  SemanticsContext &context_;
   parser::Messages &messages_;
```

Comment:
Now that this class has `context_` it probably doesn't need `messages_`. Replace `messages_.Say()` with `context_.Say()` and other occurrences of `messages_` with `context_.messages()`.

Reply:
Actually, there were a few classes that were using parser::Messages.  I've purged all of them from the file in favor of SemanticsContext.  Thanks for pointing out this simplification opportunity.